src/set_theory/surreal.lean

```diff
@@ -198,6 +202,166 @@ theorem numeric_nat : Π (n : ℕ), numeric n
 theorem numeric_omega : numeric omega :=
 ⟨by rintros ⟨⟩ ⟨⟩, λ i, numeric_nat i.down, by rintros ⟨⟩⟩
 
+theorem zero_lt_one : (0 : pgame) < 1 :=
+begin
+  rw lt_def,
+  left,
+  use ⟨punit.star, by split; rintro ⟨ ⟩⟩,
+end
+
+/-- The pre-game `half` is defined as `{0 | 1}`. -/
+def half : pgame := ⟨punit, punit, 0, 1⟩
+
+@[simp] lemma half_move_left : half.move_left punit.star = 0 := rfl
+
+@[simp] lemma half_move_right : half.move_right punit.star = 1 := rfl
+
+/-- The pre-game `half` is numeric. -/
+theorem numeric_half : numeric half :=
+begin
+  split,
+  { rintros ⟨ ⟩ ⟨ ⟩,
+    exact zero_lt_one },
+  split; rintro ⟨ ⟩,
+  { exact numeric_zero },
+  { exact numeric_one }
+end
+
+theorem zero_lt_half : 0 < half :=
+begin
+  rw lt_def,
+  left,
+  use punit.star,
+  split; rintro ⟨ ⟩,
+end
+
+theorem half_lt_one : half < 1 :=
+begin
+  rw lt_def,
+  right,
+  use punit.star,
+  split; rintro ⟨ ⟩,
+  exact zero_lt_one,
+end
+
+theorem add_half_self_equiv_one : half + half ≈ 1 :=
+begin
+  split; rw le_def; split,
+  { rintro (⟨⟨ ⟩⟩ | ⟨⟨ ⟩⟩),
+    { right,
+      use (sum.inr punit.star),
+      calc ((half + half).move_left (sum.inl punit.star)).move_right (sum.inr punit.star)
+          = (half.move_left punit.star + half).move_right (sum.inr punit.star) : by fsplit
+      ... = (0 + half).move_right (sum.inr punit.star) : by fsplit
+      ... ≈ 1 : zero_add_equiv 1
+      ... ≤ 1 : pgame.le_refl 1 },
+    { right,
+      use (sum.inl punit.star),
+      calc ((half + half).move_left (sum.inr punit.star)).move_right (sum.inl punit.star)
+          = (half + half.move_left punit.star).move_right (sum.inl punit.star) : by fsplit
+      ... = (half + 0).move_right (sum.inl punit.star) : by fsplit
+      ... ≈ 1 : add_zero_equiv 1
+      ... ≤ 1 : pgame.le_refl 1 } },
+  { rintro ⟨ ⟩ },
+  { rintro ⟨ ⟩,
+    left,
+    use (sum.inl punit.star),
+    calc 0 ≤ half : le_of_lt numeric_zero numeric_half zero_lt_half
+    ... ≈ 0 + half : (zero_add_equiv half).symm
+    ... = (half + half).move_left (sum.inl punit.star) : by fsplit },
+  { rintro (⟨⟨ ⟩⟩ | ⟨⟨ ⟩⟩),
+    { left,
+      use (sum.inr punit.star),
+      calc 1 ≤ 1 : pgame.le_refl 1
+          ... ≈ 1 + 0 : (add_zero_equiv 1).symm },
```

Comment:
Is there a `le_of_equiv` statement? Perhaps use that outside the `calc` block. It reads a bit strangely at the moment.

Reply:
I don't think there's an `le_of_equiv`. Nevertheless I managed to replace the `calc` proof with a nicer looking term proof.

Reply:
I dunno how to link the newer proof (which is now in a different line) here but here's the code

```lean
    { exact ⟨sum.inr punit.star, le_of_le_of_equiv (pgame.le_refl _) (add_zero_equiv _).symm⟩ },
    { exact ⟨sum.inl punit.star, le_of_le_of_equiv (pgame.le_refl _) (zero_add_equiv _).symm⟩ }
```